aws/data_source_aws_api_gateway_stage_test.go

```diff
@@ -0,0 +1,96 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+)
+
+func TestAccDataSourceAwsApiGatewayStage(t *testing.T) {
+	rName := acctest.RandString(8)
+	deploymentName := "aws_api_gateway_deployment.test"
+	dataSourceName := "data.aws_api_gateway_stage.example"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceAwsApiGatewayStageConfig(rName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrPair(deploymentName, "id", dataSourceName, "current_deployment_id"),
+				),
+			},
+		},
+	})
+}
+
+func testAccDataSourceAwsApiGatewayStageConfig(r string) string {
+	return fmt.Sprintf(`
+resource "aws_api_gateway_rest_api" "test" {
+  name = "tf-acc-test-%s"
+}
+
+resource "aws_api_gateway_resource" "test" {
+  rest_api_id = "${aws_api_gateway_rest_api.test.id}"
+  parent_id   = "${aws_api_gateway_rest_api.test.root_resource_id}"
+  path_part   = "test"
+}
+
+resource "aws_api_gateway_method" "test" {
+  rest_api_id   = "${aws_api_gateway_rest_api.test.id}"
+  resource_id   = "${aws_api_gateway_resource.test.id}"
+  http_method   = "GET"
+  authorization = "NONE"
+}
+
+resource "aws_api_gateway_method_response" "error" {
+  rest_api_id = "${aws_api_gateway_rest_api.test.id}"
+  resource_id = "${aws_api_gateway_resource.test.id}"
+  http_method = "${aws_api_gateway_method.test.http_method}"
+  status_code = "400"
+}
+
+resource "aws_api_gateway_integration" "test" {
+  rest_api_id = "${aws_api_gateway_rest_api.test.id}"
+  resource_id = "${aws_api_gateway_resource.test.id}"
+  http_method = "${aws_api_gateway_method.test.http_method}"
+
+  type                    = "HTTP"
+  uri                     = "https://www.google.co.uk"
+  integration_http_method = "GET"
+}
+
+resource "aws_api_gateway_integration_response" "test" {
+  rest_api_id = "${aws_api_gateway_rest_api.test.id}"
+  resource_id = "${aws_api_gateway_resource.test.id}"
+  http_method = "${aws_api_gateway_integration.test.http_method}"
+  status_code = "${aws_api_gateway_method_response.error.status_code}"
+}
+
+resource "aws_api_gateway_deployment" "test" {
+  depends_on = ["aws_api_gateway_integration.test"]
+
+  rest_api_id = "${aws_api_gateway_rest_api.test.id}"
+  stage_name  = "dev"
+  description = "This is a dev env"
+
+  variables = {
+    "a" = "2"
+  }
+}
+
+resource "aws_api_gateway_stage" "example" {
+	stage_name = "prod"
+	rest_api_id = "${aws_api_gateway_rest_api.test.id}"
+	deployment_id = "${aws_api_gateway_deployment.test.id}"
+}
+
+data "aws_api_gateway_stage" "example" {
+  rest_api_id = "${aws_api_gateway_rest_api.test.id}"
+  stage_name 	= "${basename(aws_api_gateway_stage.example.invoke_url)}"
```

Comment:
Can we use `stage_name = "${aws_api_gateway_stage.example.stage_name}"` instead?

Reply:
```
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./... -v -parallel 20 -run=TestAccDataSourceAwsApiGatewayStage -timeout 120m
?   	github.com/terraform-providers/terraform-provider-aws	[no test files]
=== RUN   TestAccDataSourceAwsApiGatewayStage
=== PAUSE TestAccDataSourceAwsApiGatewayStage
=== CONT  TestAccDataSourceAwsApiGatewayStage
--- PASS: TestAccDataSourceAwsApiGatewayStage (27.37s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	27.411s
```
Just to note that Acceptance tests still pass :)